src/TimeoutCancellation.php

```diff
@@ -17,10 +17,11 @@ final class TimeoutCancellation implements Cancellation
     private readonly Cancellation $cancellation;
 
     /**
-     * @param float  $timeout Seconds until cancellation is requested.
-     * @param string $message Message for TimeoutException. Default is "Operation timed out".
+     * @param float  $timeout     Seconds until cancellation is requested.
+     * @param string $message     Message for TimeoutException. Default is "Operation timed out".
+     * @param bool   $unreference Whether to unreference the timer.
      */
-    public function __construct(float $timeout, string $message = "Operation timed out")
+    public function __construct(float $timeout, string $message = "Operation timed out", bool $unreference = true)
```

Comment:
I really dislike any boolean parameter that's not purely a setter. Should probably be a separate method?

Reply:
We did use a similar parameter in `delay`. IMO, if we have a boolean parameter here it should match `delay`, being called `$reference` and default to false.